shared/nav.desktop.js

```diff
@@ -163,6 +164,10 @@ class Nav extends Component<void, Props, State> {
   }
 
   shouldComponentUpdate (nextProps, nextState) {
+    if (this.props.folderBadge !== nextProps.folderBadge) {
```

Comment:
This tripped me up, couldn't work out why render() wasn't being called.  @chrisnojima points out we could have some kind of generic comparator to avoid special casing each thing we use for badging.